lib/stat_tracker.rb

```diff
@@ -96,30 +80,36 @@ def count_of_teams
     @teams.length
   end
 
-  def average_goals_by_team(team_id, hoa = nil)
-    goals = 0
-    games = 0
+  def average_goals_by_team(team_id, hoa = nil) # game_teams?
+    goals = total_games_and_goals_by_team(team_id, hoa)[0]
+    games = total_games_and_goals_by_team(team_id, hoa)[1]
+    return 0 if games == 0
+    (goals.to_f / games.to_f).round(2) # average
+  end
+
+  def total_games_and_goals_by_team(team_id, hoa)
+    goals_games = [0, 0]
     @game_teams.each do |game_team|
-      if hoa != nil
-        if game_team.team_id == team_id && game_team.hoa == hoa
-          goals += game_team.goals
-          games += 1
-        end
-      elsif game_team.team_id == team_id
-          goals += game_team.goals
-          games += 1
+      if hoa && game_team.team_id == team_id && game_team.hoa == hoa
+        add_goals_and_games(goals_games, game_team)
+      elsif !hoa && game_team.team_id == team_id
+        add_goals_and_games(goals_games, game_team)
       end
     end
-    return 0 if games == 0
-    (goals.to_f / games.to_f).round(2)
+    goals_games
   end
 
-  def unique_team_ids
-    @game_teams.uniq { |game_team| game_team.team_id }.map{ |game_team| game_team.team_id }
+  def add_goals_and_games(goals_games, game_team)
+    goals_games[0] += game_team.goals
+    goals_games[1] += 1
```

Comment:
if we're incrementing by goals each time, can just call games.length at the end instead